azkaban-hadoop-security-plugin/src/main/java/azkaban/hive/IMetaStoreClientFactory.java

```diff
@@ -37,54 +35,37 @@
 
 
 /**
- * An implementation of {@link BasePooledObjectFactory} for {@link IMetaStoreClient}.
+ * Factory for creating an {@link IMetaStoreClient} using the {@link RetryingMetaStoreClient}
  */
-public class HiveMetaStoreClientFactory extends BasePooledObjectFactory<IMetaStoreClient> {
+public class IMetaStoreClientFactory {
 
-  private static final Logger log = LoggerFactory.getLogger(HiveMetaStoreClientFactory.class);
+  private static final Logger log = LoggerFactory.getLogger(IMetaStoreClientFactory.class);
   private final HiveConf hiveConf;
 
-  public HiveMetaStoreClientFactory(final HiveConf hiveConf) {
+  public IMetaStoreClientFactory(final HiveConf hiveConf) {
     this.hiveConf = hiveConf;
   }
 
-  private IMetaStoreClient createMetaStoreClient() throws MetaException {
+  public IMetaStoreClient create() {
```

Comment:
Did you consider making HiveConf a parameter to create()? That way you can work with just one Factory instance.